apm-sniffer/apm-sdk-plugin/mqtt-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mqtt/v3/define/MqttConsumerInstrumentation.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.mqtt.v3.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static org.apache.skywalking.apm.agent.core.plugin.match.HierarchyMatch.byHierarchyMatch;
+
+public class MqttConsumerInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "org.eclipse.paho.client.mqttv3.MqttCallback";
+    private static final String ON_SUCCESS_ENHANCE_METHOD = "messageArrived";
+    private static final String ON_SUCCESS_INTERCEPTOR = "org.apache.skywalking.apm.plugin.mqtt.v3.MqttConsumerInterceptor";
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[0];
+    }
+
+    @Override
+    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        return new InstanceMethodsInterceptPoint[] {
+            new InstanceMethodsInterceptPoint() {
+                @Override
+                public ElementMatcher<MethodDescription> getMethodsMatcher() {
+                    return named(ON_SUCCESS_ENHANCE_METHOD);
+                }
+
+                @Override
+                public String getMethodsInterceptor() {
+                    return ON_SUCCESS_INTERCEPTOR;
+                }
+
+                @Override
+                public boolean isOverrideArgs() {
+                    return false;
+                }
+            }
+        };
+    }
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return byHierarchyMatch(new String[] {ENHANCE_CLASS});
```

Comment:
Redundant array creation for calling varargs method

Reply:
How to do that?
eg:
org.apache.skywalking.apm.plugin.kafka.define.CallbackInstrumentation#enhanceClass
org.apache.skywalking.apm.plugin.hystrix.v1.define.HystrixCommandInstrumentation#enhanceClass
....

Reply:
```suggestion
        return byHierarchyMatch(ENHANCE_CLASS);
```

`byHierarchyMatch` originally only accepts an array as argument(like what you do now and your examples), now we refactor it to accept variable arguments so that you don't need to create an array explicitly, anyway, it's not a big deal :)

Reply:
Thank you for your review. I'll change it right away, which will help me write good code